source/whats-new.txt

```diff
@@ -22,6 +22,293 @@ What's New in 4.0
+For more information on ``GridFS``, see the :manual:`mongodb manual </core/gridfs>`.
+
+Below are some snippets that represent equivalent operations.
+
+Construction
+^^^^^^^^^^^^
+
+.. example::
+
+   .. code-block:: javascript
+
+      // old way
+      const gs = new GridStore(db, filename, mode[, options])
+      // new way
+      const bucket = new GridFSBucket(client.db('test')[,options])
+
+File Seeking
+^^^^^^^^^^^^
+
+GridFSBucket uses the Node.js Stream API. You can replicate file seeking
+by using the ``start`` and ``end`` options, creating a download stream
+from your ``GridFSBucket``.
+
+.. example::
+
+   .. code-block:: js
+
+      bucket.openDownloadStreamByName(filename, { start: 0, end: 100 })
+
+File Upload & Download
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. example::
+
+   .. code-block:: language
+
+      await client.connect();
+      const filename = 'test.txt'; // whatever local file name you want
+      const db = client.db();
+      const bucket = new GridFSBucket(db);
+
+      fs.createReadStream(filename)
+        .pipe(bucket.openUploadStream(filename))
+        .on('error', console.error)
+        .on('finish', () => {
+          console.log('done writing to db!');
+
+          bucket
+            .find()
+            .toArray()
+            .then(files => {
+              console.log(files);
+
+              bucket
+                .openDownloadStreamByName(filename)
+                .pipe(fs.createWriteStream('downloaded_' + filename))
+                .on('error', console.error)
+                .on('finish', () => {
+                  console.log('done downloading!');
+                  client.close();
+                });
+            });
+        });
+
+.. note::
+
+   ``GridFSBucket`` does not need to be closed like ``GridStore``.
+
+File Deletion
+^^^^^^^^^^^^^
+
+.. example::
+
+   .. code-block:: js
+
+      // old way
+      GridStore.unlink(db, name, callback);
+      // new way
+      bucket.delete(file_id);
+
+Finding File Metadata
+^^^^^^^^^^^^^^^^^^^^^
+
+File metadata that used to be accessible on the GridStore instance can be
+found by querying the bucket.
+
+.. example::
+
+   .. code-block:: js
+
+      const fileMetaDataList: GridFSFile[] = bucket.find({}).toArray();
+
+Unified Topology
+++++++++++++++++
+
+- We internally now only manage a ``unifiedTopology`` when you connect
+  to a ``mongod``. The differences between this and previous versions
+  is `detailed here <https://mongodb.github.io/node-mongodb-native/3.6/reference/unified-topology/>`__.
+
+- You must use the new ``directConnection`` :node-api-4.0:`option </mongoclientoptions.html#directconnection>`
+  to connect to unitiliazed replica set members.
+
+Explain
++++++++
+
+- Support has been added for fine-grained verbosity modes. You can
+  learn more about each mode :manual:`here </reference/method/cursor.explain/#behavior/>`.
+
+``instrumment()`` Removal
++++++++++++++++++++++++++
+
+The ``instrument()`` method has been removed.
```

Comment:
Mention that command monitoring is the appropriate replacement for this we wrote:

- Use command monitoring: `client.on('commandStarted', (ev) => {})`

but linking to the command monitoring docs here would be great